### PR TITLE
Fix vtkImageSlice typescript definitions

### PR DESCRIPTION
**Pull request by jadh4v**
Addresses issue: https://github.com/Kitware/vtk-js/issues/2878

Fix typescript definition error when using `vtkImageResliceMapper` with `vtkImageSlice`. Change `vtkImageSlice` to accept `vtkAbstractImageMapper` when calling `setMapper`.

Fixes errors such as:
```
Argument of type 'vtkImageResliceMapper' is not assignable to parameter of type
'vtkImageMapper'. Type 'vtkImageResliceMapper' is missing the following
properties from type 'vtkImageMapper': getSliceAtPosition, getClosestIJKAxis,
getBoundsForSlice, getRenderToRectangle, and 15 more.
```

### Changes

- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

- [x] ~This change adds or fixes unit tests~ 
- [x] Tested environment:
  - **vtk.js**:  latest master
  - **OS**:  Windows 11 Pro
  - **Browser**:  MS Edge Version 115.0.1901.203 (Official build) (64-bit)

### CODE REVIEW

**Review by daker — APPROVED** [review]
LGTM!

File: Sources/Rendering/Core/ImageSlice/index.js [hunk 1]
```diff
@@ -101,6 +101,11 @@ function vtkImageSlice(publicAPI, model) {
   };
 
   publicAPI.getBoundsForSlice = (slice, thickness) => {
+    if (!model.mapper?.getBoundsForSlice) {
+      vtkErrorMacro(
+        `Currently set mapper does not define method 'getBoundsForSlice()'`
+      );
+    }
```
Comment (floryst) [inline Sources/Rendering/Core/ImageSlice/index.js:108]: What about adding a getBoundsForSlice impl to ImageResliceMapper that calls vtkErrorMacro + returns uninitialized/invalid bounds?

Comment (jadh4v) [reply Sources/Rendering/Core/ImageSlice/index.js:108]: Or should that be in the `AbstractImageMapper` basically saying that derived classes should implement it?

Comment (floryst) [reply Sources/Rendering/Core/ImageSlice/index.js:108]: Ah yes, that's a good idea. Let's do that.

**Review by floryst — APPROVED** [review]

**Pull request merged**